services/pokemonService.ts

```diff
@@ -0,0 +1,149 @@
+import axios from 'axios';
+import {Pokemon} from '../App';
+import {Dispatch, SetStateAction} from 'react';
+import {Ability} from '../components/PokemonDetails';
+import {MarkerType} from '../components/Map';
+
+type EffectEntry = {
+  effect: string;
+  language: {
+    name: string;
+  };
+};
+
+export type ClickCords = {
+  coordinate: {
+    latitude: number;
+    longitude: number;
+  };
+  position: {
+    x: number;
+    y: number;
+  };
+};
+
+export const POKEMON_FETCH_LIMIT = 20;
+
+export const fetchPokemons = (
+  limit: number,
+  setPokemons: Dispatch<SetStateAction<Pokemon[]>>,
+) => {
+  axios
+    .get(`https://pokeapi.co/api/v2/pokemon?limit=${limit}`)
+    .then(response => {
+      response.data.results.map((pokemon: Pokemon) => {
+        axios
+          .get(pokemon.url)
+          .then(response => {
+            pokemon.photoUrl = response.data.sprites.front_default;
+          })
+          .catch(error => {
+            console.log(error);
+          });
+      });
+
+      setPokemons(response.data.results);
+    })
+    .catch(error => {
+      console.log(error);
+    });
+};
+
+export const fetchMorePokemons = (
```

Comment:
I think you can modify this function to use as `fetchPokemon` as well. Seem strange to have two separate functions.